google-api-go-generator/gen.go

```diff
@@ -498,6 +498,42 @@ func (a *API) apiBaseURL() string {
 	return resolveRelative(base, rel)
 }
 
+// apiBaseURLTemplate returns the value returned by apiBaseURL with the
+// following changes: 1) The hostname (excluding subdomains) is replaced with %s
+// for universe domain substitution. 2) If the value does not have a scheme,
+// https:// is prepended.
+func (a *API) apiBaseURLTemplate() (string, error) {
+	baseURL := a.apiBaseURL()
+	return replaceDomain(baseURL, "%s")
+}
+
+// replaceDomain detects the domain (excluding subdomains) in the provided
+// baseURL and replaces it with newDomain. If baseURL is empty, empty is
+// returned. If baseURL is an invalid URL, the error from url.Parse is returned.
+// If baseURL is valid but does not have a scheme, https:// is prepended to the
+// return value.
+func replaceDomain(baseURL, newDomain string) (string, error) {
```

Comment:
Should we perform the replacement if the GDU (googleapis.com) isn't in the baseURL?

I'm considering two cases:
* users who might be using the generator with non-google APIs
* alphabet/google services that don't support universes

It might be better to limit templating to "trusted" domains, which for our case is simple googleapis.com at present.

Reply:
Good point. I don't know. @viacheslav-rostovtsev any comment on this?

Reply:
Per @viacheslav-rostovtsev:  if the generator receives a default endpoint like `ths-prod.googleplex.com/_ah/api/tshealth/v1/`, the default endpoint template should be the unchanged value, `ths-prod.googleplex.com/_ah/api/tshealth/v1/`. (And not `https://ths-prod.UNIVERSE_DOMAIN/_ah/api/tshealth/v1/`.) Only `googleapis.com` should be replaced.

Reply:
Updated to only replace GDU (googleapis.com) in the baseURL, thanks!